dpdispatcher/dpcloudserver/client.py

```diff
@@ -94,22 +94,37 @@ def _login(self):
         # print(self.token)
         self.user_id = resp['user_id']
 
-    def refresh_token(self):
+    def refresh_token(self, retry=3):
         url = '/account/login'
         post_data = {
             'email': self.config['email'],
             'password': self.config['password']
         }
-        ret = requests.post(
-            urljoin(API_HOST, url),
-            json=post_data,
-            timeout=HTTP_TIME_OUT,
-        )
-        ret = json.loads(ret.text)
-        if ret['code'] == RETCODE.OK or ret['code'] == 0:
-            self.token = ret['data']['token']
-            return
-        raise ValueError(f"{url} Error: {ret['code']} {ret.get('message', ret.get('error'))} ")
+        resp_code = None
+        err = None
+        for i in range(retry):
+            resp = requests.post(
+                urljoin(API_HOST, url),
+                json=post_data,
+                timeout=HTTP_TIME_OUT,
+            )
+            resp_code = resp.status_code
+            if not resp.ok:
+                if self.debug:
+                    print(f"login retry: {i},statusCode: {resp.status_code}")
+                try:
+                    result = resp.json()
+                    err = result.get("error")
+                except:
```

Comment:
```suggestion
                except Exception:
```

`KeyboardInterrupt` should not be caught.

Reply:
Done